In [83]:
import numpy as np
import pandas as pd
from statistics import mode

Solution "by hand":

Greg, Marcia, Peter, Jan, Bobby, Cindy, and Oliver are 14, 12, 11, 10, 8, 6, and 8 years old, respectively, on their first appearances on the show. 


In [84]:
brady_bunch = pd.DataFrame(index = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver'])
brady_bunch['age'] = [14, 12, 11, 10, 8, 6, 8]

1 

The sum of their ages is:

14 + 12 + 11 + 10 + 8 + 6 + 8 = 

   26   +    21   +     22    =
   
             69     years total
             
mean is the sum over the total number of kids, or 69 over 7 ~= 9.9. The mode is 8 since it appears the most, 2 times. The median is 10, since there are three ages above 10 and three below.
The variance is a little tricky to calculate by hand, but we'll round the mean up to 10 since it's pretty close anyways and it'll be so much easier to calculate.

Variance = v = 

( (14 - 10)^2 + (12 - 10)^2 + (11 - 10) ^ 2 + (10 - 10) ^ 2 + 2((8 - 10) ^ 2) + (6 - 10) ^ 2 ) / (7 - 1) = 

( 4^2 + 2^2 + 1^2 + 0 + 2x((-2)^2) + (-4)^2 ) / 6 =

( 16 + 4 + 1 + 2x4 + 16 ) / 6 =

45 / 6 = 7.5

standard deviation = std =

sqrt(v) =

sqrt(7.5) ~= 2.5

and, finally, standard error:

se = std / sqrt(n) = 

2.5 / sqrt(7) ~= 1.0

Compared to Python's results:

In [85]:
print('sum is {}'.format(np.sum(brady_bunch['age'])))
print('mean is {}'.format(np.mean(brady_bunch['age'])))
print('median is {}'.format(np.median(brady_bunch['age'])))
print('mode is {}'.format(mode(brady_bunch['age'])))
print('variance is {}'.format(np.var(brady_bunch['age'])))
print('standard deviation is {}'.format(np.std(brady_bunch['age'], ddof=1)))
print('standard deviation of population is {}'.format(np.std(brady_bunch['age'], ddof=0)))
print('standard error is {}'.format(np.std(brady_bunch['age'], ddof=1) / np.sqrt(len(brady_bunch['age']))))


sum is 69
mean is 9.857142857142858
median is 10.0
mode is 8
variance is 6.408163265306122
standard deviation is 2.734262327610589
standard deviation of population is 2.5314350209527645
standard error is 1.0334540197243192


2 

I wouldn't choose mode as central tendancy since 8 is the mode, and it doesn't seem to represent well the central tendancy. The median and mean are close so I'll go with the median, which is a nicer whole integer 10.
    As for variance, I think the standard deviation of the population is more appropriate. If our population is the brady bunch kids, our sample contains all of the kids. So, 2.5, or the std of population is my choice measurement for variance.
    "10 +- 1 years old" only represents two of the seven kids, so standard error doesn't seem like a good measure of variance either.
    

3

Cindy has a birthday and turns 7. That pushes our mean age up to 10, keeps our median and mode the same, and lowers our variance:

In [86]:
brady_bunch.loc['Cindy', 'age'] = 7
brady_bunch

age
Greg     14
Marcia   12
Peter    11
Jan      10
Bobby     8
Cindy     7
Oliver    8

In [87]:
print('sum is {}'.format(np.sum(brady_bunch['age'])))
print('mean is {}'.format(np.mean(brady_bunch['age'])))
print('median is {}'.format(np.median(brady_bunch['age'])))
print('mode is {}'.format(mode(brady_bunch['age'])))
print('variance is {}'.format(np.var(brady_bunch['age'])))
print('standard deviation is {}'.format(np.std(brady_bunch['age'], ddof=1)))
print('standard deviation of population is {}'.format(np.std(brady_bunch['age'], ddof=0)))
print('standard error is {}'.format(np.std(brady_bunch['age'], ddof=1) / np.sqrt(len(brady_bunch['age']))))

sum is 70
mean is 10.0
median is 10.0
mode is 8
variance is 5.428571428571429
standard deviation is 2.516611478423583
standard deviation of population is 2.32992949004287
standard error is 0.9511897312113418


4

If we replace cousin Oliver with 1-year old Jessica, I expect we'll see a drop in mean, an increase in variance, but again median and mode will not change, which is actually ok because the set of kids, even if they include Jessica, still mostly represent kids 'around age 10'. Let's see:

In [88]:
brady_bunch = pd.DataFrame(index = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Jessica'])
brady_bunch['age'] = [14, 12, 11, 10, 8, 7, 1]

In [89]:
print('sum is {}'.format(np.sum(brady_bunch['age'])))
print('mean is {}'.format(np.mean(brady_bunch['age'])))
print('median is {}'.format(np.median(brady_bunch['age'])))
#print('mode is {}'.format(mode(brady_bunch['age']))) raises a statistics error: 'no unique mode'
print('no mode')
print('variance is {}'.format(np.var(brady_bunch['age'])))
print('standard deviation is {}'.format(np.std(brady_bunch['age'], ddof=1)))
print('standard deviation of population is {}'.format(np.std(brady_bunch['age'], ddof=0)))
print('standard error is {}'.format(np.std(brady_bunch['age'], ddof=1) / np.sqrt(len(brady_bunch['age']))))

sum is 63
mean is 9.0
median is 10.0
no mode
variance is 15.428571428571429
standard deviation is 4.242640687119285
standard deviation of population is 3.927922024247863
standard error is 1.6035674514745462


See that mean dropped a full year of age?  variance has also increased a lot, as well as our standard deviation, which makes sense given Jessica's age is so far from all the others. 

5

To tackle the question of "how many americans were fans of The Brady Bunch on the 50th anniversary of the show," we'll just take the limited data we have as a set with four datapoints:

TV Guide 20% fans 

Entertainment Weekly 23% 

fans Pop Culture Today 17% 

fans SciPhi Phanatic 5% fans



In [90]:
fans = pd.DataFrame(index=['TV Guide', 'Entertainment Weekly', 'Pop Culture Today', 'SciPhi Phanatic'])
fans['poll results'] = [20, 23, 17, 5]

In [91]:
print('sum is {}'.format(np.sum(fans['poll results'])))
print('mean is {}'.format(np.mean(fans['poll results'])))
print('median is {}'.format(np.median(fans['poll results'])))
#print('mode is {}'.format(mode(fans['poll results']))) raises a statistics error: 'no unique mode'
print('no mode')
print('variance is {}'.format(np.var(fans['poll results'])))
print('standard deviation is {}'.format(np.std(fans['poll results'], ddof=1)))
print('standard deviation of population is {}'.format(np.std(fans['poll results'], ddof=0)))
print('standard error is {}'.format(np.std(fans['poll results'], ddof=1) / np.sqrt(len(fans['poll results']))))

sum is 65
mean is 16.25
median is 18.5
no mode
variance is 46.6875
standard deviation is 7.88986691902975
standard deviation of population is 6.832825184358224
standard error is 3.944933459514875


Bearing in mind the fact that we have a very small sample size, we can easily just take the mean and standard error and say that probably 16 % +- 4% of the US population - somewhere between 12% and 20% - are fans of the show 50 years later.

Or, we could say that the SciPhans readers are outliers and probably don't represent the population as a whole very well. In that case, we could look at the median and say the result is closer to 19%, which agrees with the mean and error margin. 

However, I don't know exactly how to put margins on the figure, how the variance relates to the median. To be safe, we'll go with 16 % +- 4% of the US population as our final answer.